[dawn, skia] Update to 20251202.213730 and 144

ayeteadoe opened the pull request:
Summary of key vcpkg-related changes for this new skia version:
- Dawn needed to be updated for `skia[dawn]` to build
- https://github.com/google/skia/commit/410256218592fb34586325b4e065a0325256e868 fixed `SKCMS_API` to properly support skcms symbol import/export when building skia as a shared library.
- Removed manually defining `SKIA_IMPLEMENTATION` in `include/private/base/SkAPI.h` from the vcpkg port as the GN build properly sets `SKIA_DLL` and `SKIA_IMPLEMENTATION` for us via `is_component_build=true` and the `skia_library` config respectively.
- `SkPath` is in the process of becoming immutable, with `SkPathBuilder` becoming the new way to adjust the geometry. As of skia 144, the mutable `SkPath` API still exists behind a macro, so we provide feature that exposes the API so consumers aren't forced to immediately migrate if they want to update to this new port version.

---

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

ayeteadoe committed:
[dawn] Update to 20251202.213730

ayeteadoe committed:
[skia] Update to 144

BillyONeal commented on ports/skia/portfile.cmake:
```diff
@@ -229,6 +233,12 @@ if("dawn" IN_LIST FEATURES)
     list(APPEND required_externals dawn)
 endif()
 
+if("pathediting" IN_LIST FEATURES)
```
What is this feature we are adding that does not appear to exist upstream?

ayeteadoe replied:
In https://github.com/google/skia/commit/832f6e513d41ec0b268dc1596169312b37f7a182, a macro that hides the methods on `SkPath` that allow for the path to be mutated (e.g. see `ports/skia/example/skpathediting.cpp` added in this PR) was unconditionally defined in the GN build. This meant that updating to this new skia port version would break consumers that were using those methods.

So this feature was to prevent forcing users to update to the new alternative SkPathBuilder API. Another approach could be to just have a patch that unconditionally removes that define in the GN build so a feature is not needed.

A future skia version may remove support altogether, so not sure if it is more ideal for the vcpkg port to provide access until then, or just remove support now.

ayeteadoe committed:
[skia] de-vendor dawn via file removal instead of a patch

ayeteadoe committed:
[skia] remove pathediting feature and always expose mutable SkPath API

In skia 143, SkPath has started to migrate to have immutable geometry.

The mutable methods on SkPath are hidden when SK_HIDE_PATH_EDIT_METHODS
is defined, and 832f6e5 updated the GN build to always define it. Since
the functionality is not yet removed from skia, instead of adding a
feature that does not correspond to any upstream build config, we
instead add a patch that removes the SK_HIDE_PATH_EDIT_METHODS so
consumers of this vcpkg port version can still use the mutable SkPath
API.
